vendor/scripts/spade.js

```diff
@@ -0,0 +1,230 @@
+var Spade = function Spade() {
+	this.stack = [];
+}
+Spade.prototype = {
+	track: function(_elem) {
+		this.target = _elem;
+
+		var spade = this;
+
+		var el = document.createElement("div");
+
+		keyHook = null;
+		if(_elem.textInput && _elem.textInput.getElement) {
+			keyHook = _elem.textInput.getElement();
+		} else {
+			keyHook = _elem;
+		}
+		keyHook.addEventListener("keydown", function(_event) {spade.createEvent(spade.target)});
+		//Maybe this is needed depending on Firefox/other browsers? Duplicate non-diff events get compiled down.

```

Comment:
Yes, not sure why it doesn't work, but it seemed to sometimes lose keypresses when typing.